src/test/java/ua/team3/carsharingservice/service/impl/CarServiceImplTest.java

```diff
@@ -0,0 +1,219 @@
+import ua.team3.carsharingservice.dto.CreateCarRequestDto;
+import ua.team3.carsharingservice.mapper.CarMapper;
+import ua.team3.carsharingservice.model.Car;
+import ua.team3.carsharingservice.repository.CarRepository;
+
+@ExtendWith(MockitoExtension.class)
+class CarServiceImplTest {
+
+    private static final String BMW_BRAND = "BMW";
+    private static final Long CAR_ID = 1L;
+    private static final String TESLA_BRAND = "Tesla";
+    private static final Long SECOND_CAR_ID = 2L;
+    private static final String UPDATED_CAR_BRAND = "Audi";
+    @Mock
+    private CarRepository carRepository;
+    @Mock
+    private CarMapper carMapper;
+    @InjectMocks
+    private CarServiceImpl carService;
+
+    @Test
+    @DisplayName("Add new car with valid values")
+    void addNewCar_validData_success() {
+        CreateCarRequestDto carRequestDto = new CreateCarRequestDto();
+        carRequestDto.setDailyFee(BigDecimal.valueOf(100));
+        carRequestDto.setType(Car.CarType.SEDAN);
+        carRequestDto.setBrand(BMW_BRAND);
+        carRequestDto.setInventory(10);
+
+        CarDto carDto = new CarDto();
+        carDto.setId(CAR_ID);
+        carDto.setDailyFee(carRequestDto.getDailyFee());
+        carDto.setBrand(carRequestDto.getBrand());
+        carDto.setInventory(carRequestDto.getInventory());
+        carDto.setType(carRequestDto.getType());
+
+        Car car = new Car();
+        when(carMapper.toModel(carRequestDto)).thenReturn(car);
+        when(carRepository.save(car)).thenReturn(car);
+        when(carMapper.toDto(car)).thenReturn(carDto);
+
+        CarDto createdCar = carService.addCar(carRequestDto);
+
+        assertNotNull(createdCar);
+        assertEquals(BMW_BRAND, createdCar.getBrand());
+        assertEquals(10, createdCar.getInventory());
+        assertEquals(Car.CarType.SEDAN, createdCar.getType());
+    }
+
+    @Test
+    @DisplayName("Return list of cars")
+    void findAllCars_validData_success() {
+        Car firstCar = new Car();
+        firstCar.setId(CAR_ID);
+        firstCar.setBrand(BMW_BRAND);
+
+        Car secondCar = new Car();
+        secondCar.setId(SECOND_CAR_ID);
+        secondCar.setBrand(TESLA_BRAND);
+
+        CarDto firstCarDto = new CarDto();
+        firstCarDto.setId(CAR_ID);
+        firstCarDto.setBrand(BMW_BRAND);
+
+        CarDto secondCarDto = new CarDto();
+        secondCarDto.setId(SECOND_CAR_ID);
+        secondCarDto.setBrand(TESLA_BRAND);
+
+        List<Car> cars = List.of(firstCar, secondCar);
+        Page<Car> carPage = new PageImpl<>(cars);
+        Pageable pageable = Pageable.ofSize(1);
+
+        when(carRepository.findAll(pageable)).thenReturn(carPage);
+        when(carMapper.toDto(firstCar)).thenReturn(firstCarDto);
+        when(carMapper.toDto(secondCar)).thenReturn(secondCarDto);
+        
+        List<CarDto> result = carService.findAllCars(pageable);
+        
+        assertNotNull(result);
+        assertEquals(2, result.size());
```

Comment:
```suggestion
        assertEquals(expected.size() , result.size());
```
it will be more understandable without magic numbers